### PR TITLE
Support build runner

### PR DESCRIPTION
**Pull request by nshahan**
* Update dependencies to support angular_components: 0.9.0-alpha+3.
* Migrate Material Select and Auto-suggest examples to use Factory pattern.

### CODE REVIEW

File: pubspec.yaml [hunk 1]
```diff
@@ -3,17 +3,19 @@ description: An example app using the angular_components package.
...
-  angular: 5.0.0-alpha+2
-  angular_components: 0.9.0-alpha+2
-  angular_forms: 1.0.1-alpha+2
+  angular: 5.0.0-alpha+4
+  angular_components: 0.9.0-alpha+3
+  angular_forms: 1.0.1-alpha+4
   browser: ^0.10.0
   dart_to_js_script_rewriter: ^1.0.1
-  sass_builder: ^1.1.0
+  sass_builder: 1.1.2
```
Comment (chalin) [inline pubspec.yaml]: Might you have meant `^1.1.2` or did you want to pin this specifically to version 1.1.2?

File: pubspec.yaml [hunk 2]
```diff
@@ -3,17 +3,19 @@ description: An example app using the angular_components package.
...
+dev_dependencies:
+  webdriver: ^1.2.1
+  build_runner: ^0.7.0
+  build_web_compilers: ^0.2.0
 dependency_overrides:
   # Necessary with angular: 5.0.0-alpha+1 dependency.
   analyzer: ^0.31.0-alpha.1
-dev_dependencies:
-  webdriver: ^1.2.1
 transformers:
```
Comment (chalin) [inline pubspec.yaml]: Do we still really need the transformers, especially angular and $dart2js? Are we encouraging folks to use both the usual `pub serve|build` _and_ possibly the new build system?

File: pubspec.yaml (hunk 1 above)
Comment (nshahan) [reply pubspec.yaml]: Good catch. No I didn't mean to lock it down.

File: pubspec.yaml (hunk 2 above)
Comment (nshahan) [reply pubspec.yaml]: My plan was to leave them until the angular transformer goes away. I'll add a similar comment  to make it clear they are not used when building with build_runner.

**Pull request merged**